plugins/codecs/codecs-registry.c

```diff
@@ -4,46 +4,47 @@
   License: BSD 3-Clause (see LICENSE.txt)
 */
 
-#include "blosc-private.h"
 #include "blosc2/codecs-registry.h"
 #include "ndlz/ndlz.h"
 #include "zfp/blosc2-zfp.h"
+#include "blosc-private.h"
+#include "blosc2.h"
 
 void register_codecs(void) {
 
   blosc2_codec ndlz;
   ndlz.compcode = BLOSC_CODEC_NDLZ;
   ndlz.version = 1;
   ndlz.complib = BLOSC_CODEC_NDLZ;
-  ndlz.encoder = ndlz_compress;
-  ndlz.decoder = ndlz_decompress;
+  ndlz.encoder = &ndlz_compress;
+  ndlz.decoder = &ndlz_decompress;
```

Comment:
This is a serious fix.  I wonder why it was working before (?).

Reply:
Actually, they are equivalent. A function does not have a “value” so C compilers always interpret that kind of expressions as a pointer to the function. From the C standard:
> A _function designator_ is an expression that has function type. Except when it is the operand of the `sizeof` operator, the `_Alignof` operator, or the unary `&` operator, a function designator with _type_ ‘‘function returning type’’ is converted to an expression that has type ‘‘pointer to function returning _type_’’.

I think most C style manuals recommend the `&` to stress we store a pointer to the function, but it is not compulsory. I seem to recall the '&' is usually dropped in C++.

Reply:
Cool!  Makes sense now.